collections/item.go

```diff
@@ -0,0 +1,46 @@
+package collections
+
+import (
+	"fmt"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+// NewItem instantiates a new Item instance, given the value encoder of the item V.
+func NewItem[V any](sk storetypes.StoreKey, prefix Prefix, valueEncoder ValueEncoder[V]) Item[V] {
+	return (Item[V])(NewMap[noKey, V](sk, prefix, noKey{}, valueEncoder))
+}
+
+// Item is a type declaration based on Map
+// with a non-existent key.
+type Item[V any] Map[noKey, V]
+
+// Get gets the item, if it is not set it returns an ErrNotFound error.
+// If value decoding fails then an ErrEncoding is returned.
+func (i Item[V]) Get(ctx StorageProvider) (V, error) { return (Map[noKey, V])(i).Get(ctx, noKey{}) }
+
+// Set sets the item in the store. If Value encoding fails then an ErrEncoding is returned.
+func (i Item[V]) Set(ctx StorageProvider, value V) error {
+	return (Map[noKey, V])(i).Set(ctx, noKey{}, value)
+}
+
+// Has reports whether the item exists in the store or not.
+// Returns an error in case
+func (i Item[V]) Has(ctx StorageProvider) (bool, error) { return (Map[noKey, V])(i).Has(ctx, noKey{}) }
+
+// Remove removes the item in the store.
+func (i Item[V]) Remove(ctx StorageProvider) error { return (Map[noKey, V])(i).Remove(ctx, noKey{}) }
+
```

Comment:
```suggestion
func (i Item[V]) Get(ctx StorageProvider) (V, error) { return (Map[noKey, V])(i).Get(ctx, noKeyInstance) }

// Set sets the item in the store. If Value encoding fails then an ErrEncoding is returned.
func (i Item[V]) Set(ctx StorageProvider, value V) error {
	return (Map[noKey, V])(i).Set(ctx, noKey{}, value)
}

// Has reports whether the item exists in the store or not.
// Returns an error in case
func (i Item[V]) Has(ctx StorageProvider) (bool, error) { return (Map[noKey, V])(i).Has(ctx, noKeyInstance) }

// Remove removes the item in the store.
func (i Item[V]) Remove(ctx StorageProvider) error { return (Map[noKey, V])(i).Remove(ctx, noKeyInstance) }

var noKeyInstance = nokey{}
```